src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -5131,7 +5274,11 @@ protected override void DoAddAllProperties<T>(object obj, PSMemberInfoInternalCo
 
             foreach (DataColumn property in dataRow.Table.Columns)
             {
-                members.Add(new PSProperty(property.ColumnName, this, obj, property.ColumnName) as T);
+                T member = new PSProperty(property.ColumnName, this, obj, property.ColumnName) as T;
+                if (predicate != null || predicate(member))
```

Comment:
Maybe implement default delegate (with return false)? Seems we could  get more readable code (I see the pattern many times.).